src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> node;
+    private int size;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
     @Override
     public void add(T value) {
+        node = new Node<>(null, value,null);
+        if (head == null) {
+            head = tail = node;
+            size++;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        throwIndexOutOfBoundsAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = findNodeByIndex(index);
+        Node<T> newNode = new Node<>(current.prev, value, current);
+        current.prev = (current.prev != null) ? current.prev.next = newNode : (head = newNode);
+        size++;
     }
```

Comment:
This method contains a redundant variable `current`. You can remove it and use `findNodeByIndex(index)` directly in the `newNode` creation. Also, you should consider handling the case when `index == 0` separately to avoid null pointer exceptions.